gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,246 @@
+
+export interface ISlider3DOptions {
+    minimum?: number;
+    maximum?: number;
+    value?: number;
+    step?: number;
+}
+
+/**
+ * Class used to create a slider in 3D
+ */
+export class Slider3D extends Control3D {
+    private _sliderBarMaterial: StandardMaterial;
+    private _sliderThumbMaterial: StandardMaterial;
+    private _sliderThumb: Mesh;
+    private _sliderBar: Mesh;
+
+    private _minimum: number;
+    private _maximum: number;
+    private _value: number;
+    private _step: number;
+
+    /** Observable raised when the sldier value changes */
+    public onValueChangedObservable = new Observable<number>();
+
+    /**
+     * Creates a new slider
+     * @param name defines the control name
+     */
+    constructor(name?: string, options?: ISlider3DOptions) {
+        super(name);
+
+        this.minimum = options?.minimum ? options.minimum : SLIDER_MIN;
+        this.maximum = options?.maximum ? options.maximum : SLIDER_MAX;
+        this.step = options?.step ? options.step : SLIDER_STEP;
+        this.value = options?.value ? options.value : SLIDER_VAL;
+    }
+
+    /**
+     * Gets the mesh used to render this control
+     */
+    public get mesh(): Nullable<AbstractMesh> {
+        if (this.node) {
+            return this._sliderThumb;
+        }
+
+        return null;
+    }
+
+    /** Gets or sets minimum value */
+    public get minimum(): number {
+        return this._minimum;
+    }
+
+    public set minimum(value: number) {
+        if (this._minimum === value) {
+            return;
+        }
+
+        this._minimum = Math.max(value, SLIDER_MIN);
+        this._value = Math.max(Math.min(this._value, this._maximum), this._minimum);
+    }
+
+    /** Gets or sets maximum value */
+    public get maximum(): number {
+        return this._maximum;
+    }
+
+    public set maximum(value: number) {
+        if (this._maximum === value) {
+            return;
+        }
+
+        this._maximum = Math.max(value, this._minimum);
+        this._value = Math.max(Math.min(this._value, this._maximum), this._minimum);
+    }
+
+    /** Gets or sets step value */
+    public get step(): number {
+        return this._step;
+    }
+
+    public set step(value: number) {
+        if (this._step === value) {
+            return;
+        }
+
+        this._step = Math.max(Math.min(value, this._maximum - this._minimum), SLIDER_STEP);
+    }
+
+    /** Gets or sets current value */
+    public get value(): number {
+        return this._value;
+    }
+
+    public set value(value: number) {
+        if (this._value === value) {
+            return;
+        }
+
+        this._value = Math.max(Math.min(value, this._maximum), this._minimum);
+        this.onValueChangedObservable.notifyObservers(this._value);
+    }
+
+    protected get start(): number {
+        if (!this.node) {
+            return -SLIDER_SCALING / 2;
+        }
+
+        return this._sliderBar.position.x - this._sliderBar.scaling.y / 2;

```

Comment:
Would it be possible to re-orient the thumb/bar such that the axis are aligned, letting us write something more readable like "this._sliderBar.position.x - this._sliderBar.scaling.x / 2"?

Reply:
Yes, we can. I even found it very confusing every time I see this line.